pandas/_libs/missing.pyx

```diff
@@ -42,6 +42,34 @@ cdef:
     type cDecimal = Decimal  # for faster isinstance checks
 
 
+cpdef bint check_na_tuples_nonequal(object left, object right):
+    """
+    When we have NA in one of the tuples but not the other we have to check here,
+    because our regular checks fail before with ambigous boolean value.
+
+    Parameters
+    ----------
+    left: Any
+    right: Any
+
+    Returns
+    -------
+    True if we are dealing with tuples that have NA on one side and non NA on
+    the other side.
+
+    """
+    if not isinstance(left, tuple) or not isinstance(right, tuple):
+        return False
+
+    for left_element, right_element in zip(left, right):
```

Comment:
Should this assert that `len(left) == len(right)`? Guessing this is assumed upstream but technically `(1, NA)` and `(1,)` would return `False`

Reply:
Added. I think we could end up here with weird combinations, so better to be a bit more explicit